commands/create.go

```diff
@@ -481,6 +483,23 @@ func tlsPath(c CommandLine, flag string, defaultName string) string {
 	return filepath.Join(mcndirs.GetMachineCertDir(), defaultName)
 }
 
+func gzipEncode(customScriptContent []byte) (string, error) {
+	var b bytes.Buffer
+	gz := gzip.NewWriter(&b)
+	gz.Flush()
+	if _, err := gz.Write(customScriptContent); err != nil {
+		return "", err
+	}
+	if err := gz.Close(); err != nil {
+		return "", err
+	}
+	encoded := base64.RawStdEncoding.EncodeToString([]byte(b.Bytes()))
+
+	cmd := fmt.Sprintf("#!/usr/bin/env sh \n touch /tmp/userdata.sh \n chmod +x /tmp/userdata.sh \n " +
+		"echo %s | base64 -di | gzip -dc >> /tmp/userdata.sh \n /tmp/userdata.sh \n rm /tmp/userdata.sh", encoded)
```

Comment:
This shouldn't be needed because cloud-init will decode the contents for you if you tell it that the contents is encoded.